docs/libcudacxx/extended_api/functional/get_device_address.rst

```diff
@@ -0,0 +1,47 @@
+.. _libcudacxx-extended-api-functional-get-device-address:
+
+cuda::get_device_address
+==============================
+
+Defined in the header ``<cuda/functional>``:
+
+``cuda::get_device_address`` returns a valid pointer to a device object.
+It replaces uses of ``cudaGetSymbolAddress``, which requires an inout parameter.
+
+Example
+-------
+
+.. code:: cuda
+
+  #include <cuda/functional>
+
+  __device__ int device_object[] = {42, 1337, -1, 0};
+
+  __global__ void example_kernel(int *data) { ... }
+
+  void example()
+  {
+    {
+      T* host_address = cuda::std::addressof(device_object);
```

Comment:
can host code take the address of a `__device__`-only variable?

Reply:
The point is that on host you get a symbol to a thing that is *not* the address of the memory on device